src/main/kotlin/note/application/CreateDevice.kt

```diff
@@ -0,0 +1,22 @@
+package note.application
+
+import note.domain.Device
+import note.domain.DeviceIdentifier
+import note.domain.DeviceName
+import note.domain.DeviceRepository
+import note.domain.exceptions.AlreadyExistingDevice
+
+class CreateDevice(private val repository: DeviceRepository) {
+
+    /***
+     * Creates a new device and saves it to the DeviceRepository
+     * @throws AlreadyExistingDevice if a new device already exists
+     * @return The device instance that was saved inside the repository
+     */
+
+    fun create(id: DeviceIdentifier, name: DeviceName): Device {
```

Comment:
The CreateDevice use case should receive:
an id "primitive" -> String,
a name "primitive" -> String.

You can delegate the primitive call to the domain object call, like for example:

```kotlin
fun create(id: String, name: String) = create(DeviceIdentifier(id), DeviceName(name))
```

Be aware that, this method will throw an IllegalDeviceIdentifier if it is not an UUIDv4 id (add this restriction on the value object -> DeviceIdentifier) and IllegalDeviceName if the name is empty (you are already doing that).